src/relax/transform/few_shot_tuning.cc

```diff
@@ -0,0 +1,169 @@
+      /*task_name=*/NullOpt,
+      /*num_threads=*/num_threads,  // use all available local threads
+      /*rand_state=*/-1,            // -1 means use random seed
+      /*logger=*/nullptr);
+  task->Initialize();
+  task->search_strategy.value()->PreTuning(
+      /*max_trials=*/valid_count, /*num_trials_per_iter=*/valid_count,
+      /*design_spaces=*/task->space_generator.value()->GenerateDesignSpace(mod),
+      /*database=*/NullOpt,
+      /*cost_model=*/NullOpt);
+  while (valid_count > 0) {
+    Optional<Array<meta_schedule::MeasureCandidate>> candidates =
+        task->search_strategy.value()->GenerateMeasureCandidates();
+    if (!candidates.defined()) break;
+    Array<meta_schedule::BuilderInput> builder_inputs;
+    for (const meta_schedule::MeasureCandidate& candidate : candidates.value()) {
+      builder_inputs.push_back(meta_schedule::BuilderInput(
+          /*mod=*/candidate->sch->mod(),
+          /*target=*/target));
+    }
+    Array<meta_schedule::BuilderResult> builder_results = builder->Build(builder_inputs);
+    ICHECK_EQ(builder_results.size(), candidates.value().size());
+    int idx = 0;
+    for (const meta_schedule::BuilderResult& builder_result : builder_results) {
+      if (!builder_result->error_msg.defined()) {
+        results.push_back(candidates.value()[idx]->sch->mod());
+        valid_count--;
+      }
+      idx++;
+    }
+    if (runner.defined()) {
+      Array<meta_schedule::RunnerInput> runner_inputs;
+      int idx = 0;
+      for (const meta_schedule::BuilderResult& builder_result : builder_results) {
+        if (!builder_result->error_msg.defined()) {
+          runner_inputs.push_back(meta_schedule::RunnerInput(
+              /*artifact_path=*/builder_result->artifact_path.value(),
+              /*device_type=*/target->kind->name,
+              /*args_info=*/candidates.value()[idx]->args_info));
+        }
+        idx++;
+      }
+      Array<meta_schedule::RunnerFuture> runner_futures = runner.value()->Run(runner_inputs);
+      for (const meta_schedule::RunnerFuture& runner_future : runner_futures) {
+        meta_schedule::RunnerResult runner_result = runner_future->Result();
+        if (runner_result->error_msg.defined()) {
+          costs.push_back(1e10);
+        } else {
+          double sum = 0;
+          for (const FloatImm& cost : runner_result->run_secs.value()) {
+            sum += cost->value;
+          }
+          costs.push_back(sum / runner_result->run_secs.value().size());
+        }
+      }
+      ICHECK_EQ(costs.size(), results.size());
+    }
+  }
+  if (results.size() == 0) {
+    LOG(WARNING) << "No valid schedule found";
+    return prim_func;
+  }
+  int best_idx = 0;
+  if (runner.defined()) {
+    for (int i = 1; i < costs.size(); ++i) {
+      if (costs[i] < costs[best_idx]) {
+        best_idx = i;
+      }
+    }
+  } else {
+    best_idx = results.size() - 1;
```

Comment:
So, if we don't define runner, we pick the last one among the buildable candidates. 
Is there any benefit over using `valid_count` of 1 in this case?

Reply:
Good point, if `valid_count` is `1`, I think there's no gain / benefit to generate more candidate without benchmarking. We can fix the option to default to 1 when benchmarking is not enabled.